source/server/configuration_impl.cc

```diff
@@ -73,12 +73,10 @@ void MainImpl::initialize(const Json::Object& json) {
   if (json.hasObject("rate_limit_service")) {
     Json::ObjectPtr rate_limit_service_config = json.getObject("rate_limit_service");
     std::string type = rate_limit_service_config->getString("type");
-    if (type == "grpc_service") {
-      ratelimit_client_factory_.reset(new RateLimit::GrpcFactoryImpl(
-          *rate_limit_service_config->getObject("config"), *cluster_manager_));
-    } else {
-      throw EnvoyException(fmt::format("unknown rate limit service type '{}'", type));
-    }
+    ASSERT(type == "grpc_service");
+    ratelimit_client_factory_.reset(new RateLimit::GrpcFactoryImpl(
+        *rate_limit_service_config->getObject("config"), *cluster_manager_));
```

Comment:
you also need to fix line "throw EnvoyException(fmt::format("unsupported driver type: '{}'", type));" down below.